File: backend/src/main/scala/domains/user/objects/response/UserContributionRanklistItem.scala
package domains.user.objects.response

import domains.user.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

/** 贡献排行榜条目，包含用户身份和贡献值。 */
final case class UserContributionRanklistItem(
  user: UserIdentity,
  contribution: UserContribution
)

/** 提供贡献排行榜条目 JSON 编解码。 */
object UserContributionRanklistItem:
  given Encoder[UserContributionRanklistItem] = deriveEncoder[UserContributionRanklistItem]
  given Decoder[UserContributionRanklistItem] = deriveDecoder[UserContributionRanklistItem]

File: frontend/src/objects/user/response/UserContributionRanklistItem.ts
import type { UserContribution } from '@/objects/user/UserContribution'
import type { UserIdentity } from '@/objects/user/UserIdentity'

/** 用户贡献排行条目；只包含公开身份和后端计算的贡献值。 */
export type UserContributionRanklistItem = {
  user: UserIdentity
  contribution: UserContribution
}
